crates/rpc/src/lib.rs

```diff
@@ -18,6 +20,31 @@ impl BeerusRpc {
     }
 
     pub async fn run(self) -> Result<(SocketAddr, ServerHandle), Error> {
+        let local_spec_version =
+            self.spec_version().await.expect("Should never happen");
+
+        let remote_spec_version = self
+            .beerus
+            .starknet_client
+            .spec_version()
+            .await;
+
+        // check remote spec version
```

Comment:
The need for this comment is a hint that a function can be extracted that checks that the spec versions match. And such function can be a subject for a unit-test coverage.